controllers/validation/account_validation_controller.go

```diff
@@ -439,10 +452,108 @@ func (r *AccountValidationReconciler) Reconcile(ctx context.Context, request ctr
 			}
 		}
 	}
+
+	// check if account belongs to accountpool
+	if !account.IsBYOC() {
+		awsRegion := config.GetDefaultRegion()
+		awsSetupClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
+			SecretName: utils.AwsSecretName,
+			NameSpace:  awsv1alpha1.AccountCrNamespace,
+			AwsRegion:  awsRegion,
+		})
+		if err != nil {
+			connErr := fmt.Sprintf("unable to connect to default region %s", awsRegion)
+			reqLogger.Error(err, connErr)
+			return ctrl.Result{}, err
+		}
+
+		// Need to assume role into the cluster account
+		roleToAssume := acct.GetAssumeRole(&account)
+		awsAssumedRoleClient, _, err := acct.AssumeRole(&acct.AccountReconciler{}, reqLogger, &account, awsSetupClient, "", roleToAssume, "")
+		if err != nil {
+			reqLogger.Error(err, "Could not impersonate AWS account", "aws-account", account.Spec.AwsAccountID)
+			return utils.RequeueWithError(err)
+		}
+		// Get a list of regions enabled in the current account
+		regionsEnabledInAccount, err := awsAssumedRoleClient.DescribeRegions(&ec2.DescribeRegionsInput{
+			AllRegions: aws.Bool(false),
+		})
+		if err != nil {
+			// Retry on failures related to the slow AWS API
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == "OptInRequired" {
+					return utils.RequeueWithError(nil)
+				}
+			}
+			reqLogger.Error(err, "Failed to retrieve list of regions enabled in this account.")
+			return utils.RequeueWithError(err)
+		}
+		if account.Spec.RegionalServiceQuotas != nil {
+			if account.Status.RegionalServiceQuotas == nil {
+				err = acct.SetCurrentAccountServiceQuotas(reqLogger, &account, regionsEnabledInAccount)
+				if err != nil {
+					reqLogger.Error(err, "failed to set account service quotas")
+					return reconcile.Result{}, err
+				}
+			} else {
+				_, err = r.getServiceQuotaRequest(reqLogger, &account, awsSetupClient)
+				if err != nil {
+					reqLogger.Error(err, "failed to get account service quota")
+					return reconcile.Result{}, err
+				}
+
+			}
+		}
+	}
 	return utils.DoNotRequeue()
 }
 
-// SetupWithManager sets up the controller with the Manager.
+func (r *AccountValidationReconciler) getServiceQuotaRequest(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account, awsSetupClient awsclient.Client) (reconcile.Result, error) {
```

Comment:
Can this entire function be used in HandleNonCCSPendingVerification as well by removing the dependency on AccountValidationReconciler which I think is only being used to call statusUpdate?

Reply:
This function is being used in HandleNonCCSPendingVerification already.